packages/at_onboarding_cli/lib/src/onboard/at_onboarding_service_impl.dart

```diff
@@ -160,15 +154,64 @@ class AtOnboardingServiceImpl implements AtOnboardingService {
     }
   }
 
+  Future<Map<String, String>> _generateKeyPairs() async {
+    //generate user encryption keypair
+    logger.info('Generating encryption keypair');
+    var encryptionKeyPair = generateRsaKeypair();
+
+    //generate selfEncryptionKey
+    var selfEncryptionKey = generateAESKey();
+
+    stdout.writeln(
+        '[Information] Generating your encryption keys and .atKeys file\n');
+    //mapping encryption keys pairs to their names
+    Map<String, String> atKeysMap = <String, String>{};
+    var pkamPublicKey;
+    //generating pkamKeyPair only if authMode is keysFile
+    if (atOnboardingPreference.authMode == PkamAuthMode.keysFile) {
+      logger.info('Generating pkam keypair');
+      var pkamRsaKeypair = generateRsaKeypair();
+      atKeysMap[AuthKeyType.pkamPublicKey] =
+          pkamRsaKeypair.publicKey.toString();
+      atKeysMap[AuthKeyType.pkamPrivateKey] =
+          pkamRsaKeypair.privateKey.toString();
+      pkamPublicKey = pkamRsaKeypair.publicKey.toString();
+    } else if (atOnboardingPreference.authMode == PkamAuthMode.sim) {
+      // get the public key from secure element
+      pkamPublicKey =
+          atChops!.readPublicKey(atOnboardingPreference.publicKeyId!);
+      logger.info('pkam  public key from sim: $pkamPublicKey');
+      atKeysMap[AuthKeyType.pkamPublicKey] = pkamPublicKey;
+      // encryption key pair and self encryption symmetric key
+      // are not available to injected at_chops. Set it here
+      atChops!.atChopsKeys.atEncryptionKeyPair = AtEncryptionKeyPair.create(
+          encryptionKeyPair.publicKey.toString(),
+          encryptionKeyPair.privateKey.toString());
+      atChops!.atChopsKeys.symmetricKey = AESKey(selfEncryptionKey);
+    }
+    //Standard order of an atKeys file is ->
+    // pkam keypair -> encryption keypair -> selfEncryption key -> @sign: selfEncryptionKey[self encryption key again]
+    atKeysMap[AuthKeyType.encryptionPublicKey] =
+        encryptionKeyPair.publicKey.toString();
+    atKeysMap[AuthKeyType.encryptionPrivateKey] =
+        encryptionKeyPair.privateKey.toString();
+    atKeysMap[AuthKeyType.selfEncryptionKey] = selfEncryptionKey;
+    atKeysMap[_atSign] = selfEncryptionKey;
+
+    return atKeysMap;
+  }
+
   ///write newly created encryption keypairs into atKeys file
   Future<void> _generateAtKeysFile(Map<String, String> atKeysMap) async {
     //encrypting all keys with self encryption key
+    if (atOnboardingPreference.authMode == PkamAuthMode.keysFile) {
```

Comment:
Need an explanation somewhere of why we are doing these special things when the authMode is PkamAuthMode.keysFile